subprojects/core/src/integTest/groovy/org/gradle/api/tasks/LambdaInputsIntegrationTest.groovy

```diff
@@ -75,19 +76,24 @@ class LambdaInputsIntegrationTest extends AbstractIntegrationSpec implements Val
         """
     }
 
+    @ToBeFixedForConfigurationCache(because = "Non-serializable lambdas are not supported in configuration cache")
```

Comment:
I saw you added a link to the issue about better reporting for non-serializable lambdas. I am not sure this is necessary, given that this is not a test which would check the error message when configuration cache is on. I was referring to an issue about being able to store non-serializable lambdas in the configuration cache. In that case, this issue would go into the `because = ""` of the `ToBeFixedForConfigurationCache` annotation.

The `ToBeImplemented` annotation is more for tests which would check a future behavior, and this test doesn't seem to be that. You'd use `@Issue` here if the test checks a behavior in some issue. Though that is also not the case here.

All in all, I'd only link to an issue about "Support non-serializable lambdas with configuration caching" and then use a link to this issue in `because`.

Reply:
Would this issue work for that purpose?
https://github.com/gradle/gradle/issues/21109

Reply:
Yes, that is perfect.

Reply:
Done